parser/match_operations.go

```diff
@@ -437,21 +437,52 @@ func comparisonMatch(
 		if err := matchIndexValid(matches, amountMatch[1]); err != nil {
 			return fmt.Errorf("%w: opposite amounts comparison error", err)
 		}
-		for _, op := range matches[amountMatch[0]].Operations {
-			for _, otherOp := range matches[amountMatch[1]].Operations {
-				if err := oppositeAmounts(
-					op,
-					otherOp,
-				); err != nil {
-					return fmt.Errorf("%w: amounts not opposites", err)
-				}
-			}
+		if err := matchOperationsForOppositeComparisonValid(amountMatch[0], matches); err != nil {
```

Comment:
On further thought, I suggest we remove `matchOperationsForOppositeComparisonValid` because we are just wrapping `equalAmounts` with the conditionals it already contains (which I think is a bad practice).

```suggestion
		if err := equalAmounts(matches[amountMatch[0]].Operations); err != nil {
		  return fmt.Errorf("%w: amounts not equal within index %d match operations", err, amountMatch[0])
		}
```

Reply:
If we want to surface better errors in these cases (length 0 vs length 1), I'd suggest modifying `equalAmounts` itself: https://github.com/coinbase/rosetta-sdk-go/blob/f929bfc4a262afbbb4053570664be00053a0b657/parser/match_operations.go#L304-L306

Reply:
If we still want to allow `len(ops) == 1` to be considered equal in some cases, we could add another arg

Reply:
Still thinking if we'd want to consider operation amounts `equal` if there is only 1 op.